Document sequencer tombstones (#510)
digital-asset/docs.daml.com: docs/2.8.0/docs/canton/usermanual/manage_domain_entities.rst

   :start-after: user-manual-entry-begin: DynamicallyOnboardSequencerWithSeparateConsoles
   :end-before: user-manual-entry-end: DynamicallyOnboardSequencerWithSeparateConsoles
   :dedent:

A newly onboarded sequencer only serves events sequenced more recently than the "onboarding snapshot", or more precisely events with timestamps more recent than the

> timestamp at which the new sequencer's signing key becomes valid. If a participant (or mediator) connects to a newly onboarded sequencer too soon, the sequencer

Reviewer: The new sequencer's signing key normally becomes valid the very moment the sequencer becomes onboarded. So you don't really need this second clause in the sentence. However, what's missing here is that the some events may belong to transactions that were initiated before the sequencer was onboarded. And those events cause then such tombstones. So how about mentioning this correlation of "events" to earlier transactions? Then the maximum decision time is also motivated.

Author: Thanks, Andreas

> A newly onboarded sequencer only serves events sequenced more recently than the "onboarding snapshot", or more precisely events with timestamps more recent than the
> timestamp at which the new sequencer's signing key becomes valid. If a participant (or mediator) connects to a newly onboarded sequencer too soon, the sequencer
> subscription aborts with a "FAILED_PRECONDITION" error specifying "InvalidCounter" or "SEQUENCER_TOMBSTONE_ENCOUNTERED". In such a case, the participant or
> mediator should connect to another sequencer with a longer history of sequenced events before switching to the newly onboarded sequencer. To avoid such errors
> altogether, the best practice is to wait at least for the "maximum decision time" (the sum of the participant_response_timeout and mediator_reaction_timeout
> dynamic domain parameters with a default of 30 seconds each) before connecting nodes to a newly onboarded sequencer.

Reviewer 2: ```suggestion
A newly onboarded sequencer only serves events with timestamps more recent than the
timestamp at which the new sequencer's signing key becomes valid. If a participant (or mediator) connects to a newly onboarded sequencer too soon, the sequencer
subscription aborts with a "FAILED_PRECONDITION" error specifying "InvalidCounter" or "SEQUENCER_TOMBSTONE_ENCOUNTERED". If this occurs, the participant or
mediator should connect to another sequencer with a longer history of sequenced events before switching to the newly onboarded sequencer. To avoid such errors the best practice is to wait at least for the "maximum decision time" (the sum of the participant_response_timeout and mediator_reaction_timeout
dynamic domain parameters with a default of 30 seconds each) before connecting nodes to a newly onboarded sequencer.
```

   :start-after: user-manual-entry-begin: DynamicallyOnboardSequencerWithSeparateConsoles
   :end-before: user-manual-entry-end: DynamicallyOnboardSequencerWithSeparateConsoles
   :dedent:

A newly onboarded sequencer only serves events more recent than the "onboarding snapshot". In addition some events may belong to transactions initiated before
the sequencer was onboarded, but the sequencer is not in a position to sign such events and replaces them with "tombstones". If a participant (or mediator)
connects to a newly onboarded sequencer too soon and the subscription encounters a tombstone, the sequencer subscription aborts with a "FAILED_PRECONDITION" error
specifying "InvalidCounter" or "SEQUENCER_TOMBSTONE_ENCOUNTERED". If this occurs, the participant or mediator should connect to another sequencer with a longer

> history of sequenced events before switching to the newly onboarded sequencer. To avoid such errors the best practice is to wait at least for the "maximum decision time"

Reviewer 3: Nit: we should keep _time_ to denote points in time. Switch to "maximum decision duration" or similar ?

Author: Follow-up PR at: https://github.com/digital-asset/docs.daml.com/pull/514
